
# Post-alignment

### Setup

Just make sure the TxDb object is loaded again.

```{r, eval=TRUE, message=FALSE, warning=FALSE}
if (!suppressWarnings(require(TxDb.Mmusculus.UCSC.mm10.knownGene))) {
  source("http://bioconductor.org/biocLite.R")
  biocLite("TxDb.Mmusculus.UCSC.mm10.knownGene")
  library(TxDb.Mmusculus.UCSC.mm10.knownGene)
}
```

[GenomicAlignments](http://bioconductor.org/packages/release/bioc/html/GenomicAlignments.html) provides functions for storing, manipulating and counting short read alignments. 

```{r, eval=TRUE, message=FALSE, warning=FALSE}
if (!suppressWarnings(require(GenomicAlignments))) {
  source("http://bioconductor.org/biocLite.R")
  biocLite("GenomicAlignments")
  library(GenomicAlignments)
}
```

### QC

**Tools**:

* Picard
* Samtools
* RSeQC
* FASTQC

**Metrics**:

* % mapped
* % multi-mapping
* duplication rate
* GC bias
* coverage bias 
* replicate correlation

### Count Reads

Most differential expression analysis programs take as input raw counts of the number of reads mapping to each gene. 

It's common to perform this step on the command line with [HTseq-count](https://htseq.readthedocs.io/en/release_0.9.1/). But we can just as simply do so in R.

First extract exon locations from the txdb object again:
```{r, eval=TRUE, message=FALSE, warning=FALSE}
exons <- exonsBy(TxDb.Mmusculus.UCSC.mm10.knownGene)
exons_chr10 <- keepSeqlevels(exons, "chr10")
```

Then find the output bam files from alignment:
```{r, eval=TRUE, message=FALSE, warning=FALSE}
bam_files <- list.files(path = "bam", full.names = T)
names(bam_files) <- c("BCR", "EV")
bam_files <- BamFileList(bam_files, yieldSize=2000000)
```

Finally we count reads using the ```summarizeOverlaps()``` function with the following arguments:
    
* features = regions to count reads over (the exons in this case)
* reads = bam files to count from
* mode = determines how reads are assigned to features - see [docs](http://bioconductor.org/packages/devel/bioc/vignettes/GenomicAlignments/inst/doc/summarizeOverlaps.pdf)
* singleEnd = indicates whether reads are SE
* ignore.strand = indicates if library is stranded
* fragments = whether to include unpaired reads

```{r, eval=TRUE, message=FALSE, warning=FALSE}
se <- summarizeOverlaps(features = exons_chr10,
                        reads = bam_files,
                        singleEnd = FALSE,
                        ignore.strand = TRUE,
                        fragments = TRUE)
```

This produces a count matrix with one column per sample and one row per gene. The data is stored as a summarised experiment which can be subsequently used in differential expression analysis, e.g. with [DESeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html) or [limma](https://bioconductor.org/packages/release/bioc/html/limma.html).

Examining the se object indicates the dimensions, what assay data is stored (counts), the rownames - representing entrez IDs, columns/conditions:
```{r, eval=TRUE, message=FALSE, warning=FALSE}
se
```

Use ```assay()``` to access the counts:
```{r, eval=TRUE, message=FALSE, warning=FALSE}
head(assay(se))
```

We can sum the counts per samples with ```colSums()```:
```{r, eval=TRUE, message=FALSE, warning=FALSE}
colSums(assay(se))
```

Thats it! You're now ready for downstream analysis and interpretation of results.