<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Intro to Bioinformatics</title>
  <meta name="description" content="Intro to Bioinformatics">
  <meta name="generator" content="bookdown 0.4 and GitBook 2.6.7">

  <meta property="og:title" content="Intro to Bioinformatics" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Intro to Bioinformatics" />
  
  
  

<meta name="author" content="Mark Robinson">


<meta name="date" content="2017-09-08">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="01.2_unix.html">
<link rel="next" href="01.4_qc.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i>Contact</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="01.0_quick_start.html"><a href="01.0_quick_start.html"><i class="fa fa-check"></i><b>1</b> Quick Start Tutorial</a><ul>
<li class="chapter" data-level="1.1" data-path="01.1_setup.html"><a href="01.1_setup.html"><i class="fa fa-check"></i><b>1.1</b> System Setup</a><ul>
<li class="chapter" data-level="1.1.1" data-path="01.1_setup.html"><a href="01.1_setup.html#r"><i class="fa fa-check"></i><b>1.1.1</b> R</a></li>
<li class="chapter" data-level="1.1.2" data-path="01.1_setup.html"><a href="01.1_setup.html#rstudio"><i class="fa fa-check"></i><b>1.1.2</b> RStudio</a></li>
<li class="chapter" data-level="1.1.3" data-path="01.1_setup.html"><a href="01.1_setup.html#r-packages"><i class="fa fa-check"></i><b>1.1.3</b> R packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="01.2_unix.html"><a href="01.2_unix.html"><i class="fa fa-check"></i><b>1.2</b> <img src="images/terminal_symbol_small.png" alt="symbol" /> UNIX Essentials</a><ul>
<li class="chapter" data-level="1.2.1" data-path="01.2_unix.html"><a href="01.2_unix.html#the-terminal"><i class="fa fa-check"></i><b>1.2.1</b> The Terminal</a></li>
<li class="chapter" data-level="1.2.2" data-path="01.2_unix.html"><a href="01.2_unix.html#warning"><i class="fa fa-check"></i><b>1.2.2</b> WARNING</a></li>
<li class="chapter" data-level="1.2.3" data-path="01.2_unix.html"><a href="01.2_unix.html#the-filesystem"><i class="fa fa-check"></i><b>1.2.3</b> The Filesystem</a></li>
<li class="chapter" data-level="1.2.4" data-path="01.2_unix.html"><a href="01.2_unix.html#navigating-the-filesystem"><i class="fa fa-check"></i><b>1.2.4</b> Navigating the Filesystem</a></li>
<li class="chapter" data-level="1.2.5" data-path="01.2_unix.html"><a href="01.2_unix.html#head-tail"><i class="fa fa-check"></i><b>1.2.5</b> Head &amp; Tail</a></li>
<li class="chapter" data-level="1.2.6" data-path="01.2_unix.html"><a href="01.2_unix.html#arguments"><i class="fa fa-check"></i><b>1.2.6</b> Arguments</a></li>
<li class="chapter" data-level="1.2.7" data-path="01.2_unix.html"><a href="01.2_unix.html#inputoutput"><i class="fa fa-check"></i><b>1.2.7</b> Input/Output</a></li>
<li class="chapter" data-level="1.2.8" data-path="01.2_unix.html"><a href="01.2_unix.html#summary"><i class="fa fa-check"></i><b>1.2.8</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="01.3_data.html"><a href="01.3_data.html"><i class="fa fa-check"></i><b>1.3</b> Downloading Public Data</a><ul>
<li class="chapter" data-level="1.3.1" data-path="01.3_data.html"><a href="01.3_data.html#geo-sra"><i class="fa fa-check"></i><b>1.3.1</b> GEO &amp; SRA</a></li>
<li class="chapter" data-level="1.3.2" data-path="01.3_data.html"><a href="01.3_data.html#the-fastq-format"><i class="fa fa-check"></i><b>1.3.2</b> The FASTQ format</a></li>
<li class="chapter" data-level="1.3.3" data-path="01.3_data.html"><a href="01.3_data.html#base-quality-scores"><i class="fa fa-check"></i><b>1.3.3</b> Base Quality Scores</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="01.4_qc.html"><a href="01.4_qc.html"><i class="fa fa-check"></i><b>1.4</b> Initial quality control</a><ul>
<li class="chapter" data-level="1.4.1" data-path="01.4_qc.html"><a href="01.4_qc.html#running-fastqc"><i class="fa fa-check"></i><b>1.4.1</b> Running FastQC</a></li>
<li class="chapter" data-level="1.4.2" data-path="01.4_qc.html"><a href="01.4_qc.html#summarising-with-multiqc"><i class="fa fa-check"></i><b>1.4.2</b> Summarising with MultiQC</a></li>
<li class="chapter" data-level="1.4.3" data-path="01.4_qc.html"><a href="01.4_qc.html#interpreting-results"><i class="fa fa-check"></i><b>1.4.3</b> Interpreting Results</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="02.1_recommended.html"><a href="02.1_recommended.html"><i class="fa fa-check"></i><b>1.5</b> System Setup</a><ul>
<li class="chapter" data-level="1.5.1" data-path="02.1_recommended.html"><a href="02.1_recommended.html#operating-system"><i class="fa fa-check"></i><b>1.5.1</b> Operating system</a></li>
<li class="chapter" data-level="1.5.2" data-path="02.1_recommended.html"><a href="02.1_recommended.html#r-1"><i class="fa fa-check"></i><b>1.5.2</b> R</a></li>
<li class="chapter" data-level="1.5.3" data-path="02.1_recommended.html"><a href="02.1_recommended.html#rstudio-1"><i class="fa fa-check"></i><b>1.5.3</b> RStudio</a></li>
<li class="chapter" data-level="1.5.4" data-path="02.1_recommended.html"><a href="02.1_recommended.html#sra-toolkit"><i class="fa fa-check"></i><b>1.5.4</b> SRA-Toolkit</a></li>
<li class="chapter" data-level="1.5.5" data-path="02.1_recommended.html"><a href="02.1_recommended.html#r-packages-1"><i class="fa fa-check"></i><b>1.5.5</b> R packages</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Intro to Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="downloading-public-data" class="section level2">
<h2><span class="header-section-number">1.3</span> Downloading Public Data</h2>
<p>You might frequently want to re-analyse published data, or compare it with your results. This section runs through how to download some example data.</p>
<p>We’ll be downloading the raw RNA-seq data from our labs recent publication:</p>
<p>First lets make a folder to store the data in, we’ll call it fastq:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">mkdir</span> fastq</code></pre></div>
<div id="geo-sra" class="section level3">
<h3><span class="header-section-number">1.3.1</span> GEO &amp; SRA</h3>
<p>The most commonly used database to store expression-related data is the <a href="https://www.ncbi.nlm.nih.gov/geo/">Gene Expression Omnibus (GEO)</a> repository. GEO archives processed data and brokers the raw sequencing reads to the <a href="https://www.ncbi.nlm.nih.gov/sra">Sequence Read Archive (SRA)</a>.</p>
<p>The format of data stored on GEO is fairly un-structured and is mainly up to the submitter, so you’ll have to read the records and/or publication to determine what is actually provided. The SRA records on the other hand should always contain raw read data.</p>
<p>Unaligned read data is usually stored in the FASTQ format. To extract FASTQ data from SRA we’ll be using the <strong>fastq-dump</strong> tool from the <strong>SRA-toolkit</strong>. For this we just need the SRA run (SRR) ID, of the form ‘SRR*****’. You can find this via either the SRA or GEO record for each sample, for the current example the SRR of interest are listed below.</p>
<p>The default behaviour of the fastq-dump tool is a little odd so we need to add some extra arguments. This makes the commands below look complicated but you don’t need to worry about all the options right now - most of them just ensure you get a standard pair of FASTQ formatted files, see the full explanation <a href="">here</a> if you’re interested.</p>
<p>The important parts are at the end: <code>--outdir fastq</code> tells the program to output the FASTQ files to the ‘fastq’ directory we just made. The last argument should be the SRR ID for the chosen run.</p>
<p>Copy each command below one line at a time to your terminal and run, they should take ~?? to download and extract depending on your connection.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">fastq-dump</span> --readids --dumpbase --gzip --clip --skip-technical --read-filter pass --split-files --outdir fastq  SRR
<span class="kw">fastq-dump</span> --readids --dumpbase --gzip --clip --skip-technical --read-filter pass --split-files --outdir fastq  SRR
<span class="kw">fastq-dump</span> --readids --dumpbase --gzip --clip --skip-technical --read-filter pass --split-files --outdir fastq  SRR
<span class="kw">fastq-dump</span> --readids --dumpbase --gzip --clip --skip-technical --read-filter pass --split-files --outdir fastq  SRR</code></pre></div>
</div>
<div id="the-fastq-format" class="section level3">
<h3><span class="header-section-number">1.3.2</span> The FASTQ format</h3>
<p>Unaligned read data is usually stored in the FASTQ format, it’s text-based (i.e. human-readable) so we can take a quick look using the head command - the record for each read is (usually) composed of 4 lines, so we’ll just print the first record by specifying <code>-n 4</code>:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">head</span> -n 4 fastq/???</code></pre></div>
<p>This should have printed the following lines:</p>
<pre><code>@HISEQ:555:C8NT6ACXX:5:1101:1277:1979 1:N:0:TGACCA
NTCTGGTGTCAGGCCTCCTCACTGTGAGTCATCAGTACAGAGGCACAGGGAGGTGAGGTGGGAAGTAAATCGGGGTGCAGGAAATGGGAAGAGAATGGGA
+
#4=DDFDFHHHHHJJJJJJJJJJJIJJJHJIJJJIHIJJJJJJIIIHIJJGHI;CFHG@DHGHFHHFFFFFFDDDD?BCCDBBDDDDDCDDBDDCCDCD?</code></pre>
<p>FASTQ (usually) contains 4 lines per read:</p>
<ul>
<li>Line 1 = (unique) Read ID</li>
<li>Line 2 = The base call (A,T,C,G or N) for each position in the read</li>
<li>Line 3 = Just a “+”! (maybe the read ID again) - pretty useless, just ignore</li>
<li>Line 4 = Quality of each base call (encoded with letters/numbers/symbols)</li>
</ul>
<p>You probably don’t need to know what all the info in the read ID is right now but there’s an explanation <a href="link">here</a> if you really want to know.</p>
<p>The sequence quality is rather confusingly first <strong>Phred scaled</strong> then encoded by an ASCII character, you should know about base quality scores but learning the conversion to ASCII is pretty pointless. A general rule of thumb would be letters = high quality, numbers = medium, symbols = low quality.</p>
</div>
<div id="base-quality-scores" class="section level3">
<h3><span class="header-section-number">1.3.3</span> Base Quality Scores</h3>
<p>The base quality score indicates the probability of a base call being wrong for each position. The probablity is scaled with the equation below to produce the <strong>Phred quality score</strong>:</p>
<p><span class="math display">\[ Q = -10 log_{10} P \]</span></p>
<p>And if you hate maths: “ten times the number of zeros you need to wack in front of a 1 to get the probability”</p>
<p>Examples:</p>
<table>
<caption><span id="tab:unnamed-chunk-19">Table 1.1: </span>Converting base quality scores.</caption>
<thead>
<tr class="header">
<th align="left">Phred Quality Score</th>
<th align="left">Probability of error</th>
<th align="left">Base call accuracy</th>
<th align="left">ASCII encoding</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">10</td>
<td align="left">0.1 (1 in 10)</td>
<td align="left">90%</td>
<td align="left">+</td>
</tr>
<tr class="even">
<td align="left">20</td>
<td align="left">0.01 (1 in 100)</td>
<td align="left">99%</td>
<td align="left">5</td>
</tr>
<tr class="odd">
<td align="left">30</td>
<td align="left">0.001 (1 in 1,000)</td>
<td align="left">99.9%</td>
<td align="left">?</td>
</tr>
<tr class="even">
<td align="left">40</td>
<td align="left">0.0001 (1 in 10,000)</td>
<td align="left">99.99%</td>
<td align="left">I</td>
</tr>
</tbody>
</table>
<p>Basic take home message: higher = better.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="01.2_unix.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="01.4_qc.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
