<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Intro to Bioinformatics</title>
  <meta name="description" content="Intro to Bioinformatics">
  <meta name="generator" content="bookdown 0.4 and GitBook 2.6.7">

  <meta property="og:title" content="Intro to Bioinformatics" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Intro to Bioinformatics" />
  
  
  

<meta name="author" content="Mark Robinson">


<meta name="date" content="2017-09-08">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="01.3_data.html">
<link rel="next" href="02.1_recommended.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#contact"><i class="fa fa-check"></i>Contact</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="01.0_quick_start.html"><a href="01.0_quick_start.html"><i class="fa fa-check"></i><b>1</b> Quick Start Tutorial</a><ul>
<li class="chapter" data-level="1.1" data-path="01.1_setup.html"><a href="01.1_setup.html"><i class="fa fa-check"></i><b>1.1</b> System Setup</a><ul>
<li class="chapter" data-level="1.1.1" data-path="01.1_setup.html"><a href="01.1_setup.html#r"><i class="fa fa-check"></i><b>1.1.1</b> R</a></li>
<li class="chapter" data-level="1.1.2" data-path="01.1_setup.html"><a href="01.1_setup.html#rstudio"><i class="fa fa-check"></i><b>1.1.2</b> RStudio</a></li>
<li class="chapter" data-level="1.1.3" data-path="01.1_setup.html"><a href="01.1_setup.html#r-packages"><i class="fa fa-check"></i><b>1.1.3</b> R packages</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="01.2_unix.html"><a href="01.2_unix.html"><i class="fa fa-check"></i><b>1.2</b> <img src="images/terminal_symbol_small.png" alt="symbol" /> UNIX Essentials</a><ul>
<li class="chapter" data-level="1.2.1" data-path="01.2_unix.html"><a href="01.2_unix.html#the-terminal"><i class="fa fa-check"></i><b>1.2.1</b> The Terminal</a></li>
<li class="chapter" data-level="1.2.2" data-path="01.2_unix.html"><a href="01.2_unix.html#warning"><i class="fa fa-check"></i><b>1.2.2</b> WARNING</a></li>
<li class="chapter" data-level="1.2.3" data-path="01.2_unix.html"><a href="01.2_unix.html#the-filesystem"><i class="fa fa-check"></i><b>1.2.3</b> The Filesystem</a></li>
<li class="chapter" data-level="1.2.4" data-path="01.2_unix.html"><a href="01.2_unix.html#navigating-the-filesystem"><i class="fa fa-check"></i><b>1.2.4</b> Navigating the Filesystem</a></li>
<li class="chapter" data-level="1.2.5" data-path="01.2_unix.html"><a href="01.2_unix.html#head-tail"><i class="fa fa-check"></i><b>1.2.5</b> Head &amp; Tail</a></li>
<li class="chapter" data-level="1.2.6" data-path="01.2_unix.html"><a href="01.2_unix.html#arguments"><i class="fa fa-check"></i><b>1.2.6</b> Arguments</a></li>
<li class="chapter" data-level="1.2.7" data-path="01.2_unix.html"><a href="01.2_unix.html#inputoutput"><i class="fa fa-check"></i><b>1.2.7</b> Input/Output</a></li>
<li class="chapter" data-level="1.2.8" data-path="01.2_unix.html"><a href="01.2_unix.html#summary"><i class="fa fa-check"></i><b>1.2.8</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="01.3_data.html"><a href="01.3_data.html"><i class="fa fa-check"></i><b>1.3</b> Downloading Public Data</a><ul>
<li class="chapter" data-level="1.3.1" data-path="01.3_data.html"><a href="01.3_data.html#geo-sra"><i class="fa fa-check"></i><b>1.3.1</b> GEO &amp; SRA</a></li>
<li class="chapter" data-level="1.3.2" data-path="01.3_data.html"><a href="01.3_data.html#the-fastq-format"><i class="fa fa-check"></i><b>1.3.2</b> The FASTQ format</a></li>
<li class="chapter" data-level="1.3.3" data-path="01.3_data.html"><a href="01.3_data.html#base-quality-scores"><i class="fa fa-check"></i><b>1.3.3</b> Base Quality Scores</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="01.4_qc.html"><a href="01.4_qc.html"><i class="fa fa-check"></i><b>1.4</b> Initial quality control</a><ul>
<li class="chapter" data-level="1.4.1" data-path="01.4_qc.html"><a href="01.4_qc.html#running-fastqc"><i class="fa fa-check"></i><b>1.4.1</b> Running FastQC</a></li>
<li class="chapter" data-level="1.4.2" data-path="01.4_qc.html"><a href="01.4_qc.html#summarising-with-multiqc"><i class="fa fa-check"></i><b>1.4.2</b> Summarising with MultiQC</a></li>
<li class="chapter" data-level="1.4.3" data-path="01.4_qc.html"><a href="01.4_qc.html#interpreting-results"><i class="fa fa-check"></i><b>1.4.3</b> Interpreting Results</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="02.1_recommended.html"><a href="02.1_recommended.html"><i class="fa fa-check"></i><b>1.5</b> System Setup</a><ul>
<li class="chapter" data-level="1.5.1" data-path="02.1_recommended.html"><a href="02.1_recommended.html#operating-system"><i class="fa fa-check"></i><b>1.5.1</b> Operating system</a></li>
<li class="chapter" data-level="1.5.2" data-path="02.1_recommended.html"><a href="02.1_recommended.html#r-1"><i class="fa fa-check"></i><b>1.5.2</b> R</a></li>
<li class="chapter" data-level="1.5.3" data-path="02.1_recommended.html"><a href="02.1_recommended.html#rstudio-1"><i class="fa fa-check"></i><b>1.5.3</b> RStudio</a></li>
<li class="chapter" data-level="1.5.4" data-path="02.1_recommended.html"><a href="02.1_recommended.html#sra-toolkit"><i class="fa fa-check"></i><b>1.5.4</b> SRA-Toolkit</a></li>
<li class="chapter" data-level="1.5.5" data-path="02.1_recommended.html"><a href="02.1_recommended.html#r-packages-1"><i class="fa fa-check"></i><b>1.5.5</b> R packages</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Intro to Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="initial-quality-control" class="section level2">
<h2><span class="header-section-number">1.4</span> Initial quality control</h2>
<p>The first step once you have your FASTQ data is to check the quality of the reads. For this we’ll simply run <strong>fastqc</strong> - the standard tool for this process. It’s also super easy to run and produces a <strong>html</strong> summary per sample.</p>
<div id="running-fastqc" class="section level3">
<h3><span class="header-section-number">1.4.1</span> Running FastQC</h3>
<p>Make a new directory to store all qc info:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">mkdir</span> qc</code></pre></div>
<p>Run FASTQC on each pair of FASTQ files:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">fastqc</span> --outdir qc fastq/?? fastq/??</code></pre></div>
<p>For each input FASTQ, fastqc will produce a <strong>.zip</strong> archive and and <strong>.html</strong> you can open this html with a web browser.</p>
</div>
<div id="summarising-with-multiqc" class="section level3">
<h3><span class="header-section-number">1.4.2</span> Summarising with MultiQC</h3>
<p>You can easily summarise this data accross all samples with MultiQC - just call multiqc on the qc directory:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">multiqc</span> qc</code></pre></div>
<p>MultiQC will find any FastQC files and summarise them in interactive plots for you (it also handles a lot of other QC tool outputs too!). It should produce <strong>multiqc_report.html</strong> - open this in a web browser.</p>
</div>
<div id="interpreting-results" class="section level3">
<h3><span class="header-section-number">1.4.3</span> Interpreting Results</h3>
<p>The report contains some general stats at the top - number of sequences etc., then specific plots for various metrics.</p>
<p>Some general points:</p>
<ul>
<li>FastQC attempts to assign a pass/warn/fail for each metric - corresponding to the green/orange/red colouring in plots. * Some plots also have the range of values in the y axis coloured - green = high quality, orange = medium, red = poor.</li>
<li>You can hover your mouse over a specific line to find out what sample it corresponds to</li>
<li>The bar next to the plot title indicates the total number passing/warnings/failed for each metric - hover over it to see which samples are in each category</li>
<li>Plots and values can be saved through the toolbox on the right</li>
<li>click the <strong>FastQC help</strong> link to see details about each metric and possible reasons for failure</li>
</ul>
<p>Common issues:</p>
<ul>
<li>Sequence quality degrades with increasing read length so you’ll often see a drop towards the end of reads - particularly for longer reads</li>
<li>A temporary drop in quality in the middle of reads could indicate a run issue such as bubbles running through the flow cell</li>
<li>Over-represented sequences - such as contaminants or spike-ins can bias base composition, GC content and duplication</li>
<li>Base composition bias in the first ~12bp - this is often observed due to the fact that random hexamer priming isn’t really random, but doesn’t appear to have an effect downstream and can generally just be ignored.</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="01.3_data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="02.1_recommended.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
